src/main/java/org/beanpod/switchboard/service/StreamLogService.java

```diff
@@ -27,8 +34,25 @@ public StreamLogDto createLog(
 
     streamLog.setLogEntity(logEntity);
 
-    StreamLogDto streamLogDto = logStreamMapper.toLogStreamDto(streamLog);
+    StreamLogDto streamLogDto = streamLogMapper.toLogStreamDto(streamLog);
 
     return streamLogDao.createStreamLog(streamLogDto);
   }
+
+  public StreamLogDto createLog(CreateStreamLogRequest createStreamLogRequest) {
+    Optional<StreamDto> streamDto =
+        Optional.of(Long.valueOf(createStreamLogRequest.getStreamId()))
+            .map(streamDao::getStreamById);
+
+    if (!streamDto.isPresent()) {
+      return null;
+    }
```

Comment:
Just as a tip, try to avoid using negation (!) when possible. Here you can use `.isEmpty()`. It's something small, but it helps with readability.